ast/format.go

```diff
@@ -376,9 +376,28 @@ func (f *formatter) formatIdentifier(n *Identifier) {
 }
 
 func (f *formatter) formatStringLiteral(n *StringLiteral) {
-	f.writeRune('"')
-	f.writeString(n.Value)
-	f.writeRune('"')
+	if n.Loc != nil && n.Loc.Source != "" {
+		// Preserve the exact literal if we have it
+		f.writeString(n.Loc.Source)
+	} else {
+		f.writeRune('"')
+		f.writeString(escapeStr(n.Value))
+		f.writeRune('"')
+	}
+}
+
+func escapeStr(s string) string {
```

Comment:
From what I've seen in the Go standard library, it's more efficient to call `strings.Index` to determine if there are any problematic characters to begin with or you can use `strings.Count` which would be even better. Then if there are characters that require an escape, you use this path (and with `strings.Count` you can know the exact size).

Otherwise, you use the value directly.

Reply:
Since it saves an allocation, it seems like a good optimization. Although this is not very performance critical code. I'll make the change as it is straightforward.